numpy/random/mtrand/mtrand.pyx

```diff
@@ -4967,33 +4969,53 @@ cdef class RandomState:
                [0, 1, 2]])
 
         """
-        cdef npy_intp i, j
-
-        i = len(x) - 1
-
-        # Logic adapted from random.shuffle()
-        if isinstance(x, np.ndarray) and \
-           (x.ndim > 1 or x.dtype.fields is not None):
-            # For a multi-dimensional ndarray, indexing returns a view onto
-            # each row. So we can't just use ordinary assignment to swap the
-            # rows; we need a bounce buffer.
+        cdef:
+            npy_intp i, j, n = len(x)
+            size_t stride, nbytes
```

Comment:
We generally use `npy_intp`, not `size_t`. Note that arrays will be in a valid range for `npy_intp`.